rviz_common/src/rviz_common/uniform_string_stream.cpp

```diff
@@ -61,6 +61,11 @@ void UniformStringStream::parseFloat(float & f)
   if (float_reader.fail()) {
     this->setstate(std::ios::failbit);
   }
+
+  // Ensure the entire string is correctly parsed
+  if (float_reader.fail() || !float_reader.eof()) {
+    this->setstate(std::ios::failbit);
+  }
```

Comment:
```suggestion
  // Ensure the entire string is correctly parsed
  if (float_reader.fail() || !float_reader.eof()) {
    this->setstate(std::ios::failbit);
  }
```

Reply:
I'm not entirely familiar with this code, but would this cause `std::ios::failbit` to be set after the first `parseFloat()` call, since the string is not fully consumed after that?